### PR TITLE
CR 1117644 Generated the static libs for hw_emu and sw_emu driver cod…

### PR DESCRIPTION
**Pull request by venkatp-xilinx**
…e base

Enhancement request to have the Static HAL libs for hw_emu and sw_emu

This is an issue filed with the CR 1117644

As requested, created the static HAL libraries for hw_emu and sw_emu which helps  to have statically compiled host code for hw/sw emulation

No Risk

Verified the existing scenarios. All are looking good

No

### CODE REVIEW

**Review by chvamshi-xilinx — APPROVED** [review]
Looks good. Hope you have verified one simple test with these static libraries.

File: src/runtime_src/core/pcie/emulation/cpu_em/CMakeLists.txt [hunk 1]
```diff
@@ -46,13 +50,24 @@ target_link_libraries(xrt_swemu
   uuid
   )
 
+target_link_libraries(xrt_swemu_static
+  PRIVATE
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/cpu_em/CMakeLists.txt:54]: INTERFACE, please see comment in core/pcie/linux/CMakeLists.txt

File: src/runtime_src/core/pcie/emulation/hw_em/CMakeLists.txt [hunk 2]
```diff
@@ -50,14 +54,26 @@ target_link_libraries(xrt_hwemu
   uuid
   )
 
+target_link_libraries(xrt_hwemu_static
+  PRIVATE
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/CMakeLists.txt:58]: Ditto

File: src/runtime_src/core/pcie/emulation/common_em/CMakeLists.txt [hunk 3]
```diff
@@ -52,3 +52,9 @@ target_link_libraries(common_em
   rt
   )
 
+install(TARGETS common_em
+  EXPORT xrt-dev-targets
+  ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT}
+  LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT} ${XRT_NAMELINK_ONLY}
+)
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/common_em/CMakeLists.txt:59]: I would suggest to include common_em_objects in the static shim emulation libraries and not release common_em library.

**Review by stsoe — CHANGES_REQUESTED** [review]

**Review by stsoe — APPROVED** [review]

**Pull request merged**